### ISSUE TITLE
Uncaught TypeError: Cannot read property 'getAttribute' of null

### ISSUE THREAD

**Issue by ipankajkumar93** [HabibRh26/HabibRh26.github.io#2]
Everytime we change the section, we get this error in console.

Uncaught TypeError: Cannot read property 'getAttribute' of null

Reason : It can not get the attribute of main logo and inverted logo defined in navbar.js

let mainLogo = document.getElementById("main-logo").getAttribute("src");
 let invertedLogo = document.getElementById("inverted-logo").getAttribute("src");

![Screenshot (17)](https://user-images.githubusercontent.com/24999656/97407210-5de70080-1920-11eb-80d6-d91074729a26.png)

**Comment by HabibRh26** [comment]
@Have you checked the `static>asset>image` folder if there is any "main-logo" image resides or not?

**Comment by ipankajkumar93** [comment]
This error comes for your blog.
I opened HabibRh26.github.io
And there is no static folder in your repository.

**Comment by HabibRh26** [comment]
@ipankajsingh I removed the image which was given default by the theme author, so may be that is why. I would request you to go through the theme owner repo because as I customized it according to my need so many things may not work like you want.

**Comment by HabibRh26** [comment]
### Reply to the email that you sent to me:
ans of para1: <br>
Yes, you got it right about the navigation issue on mobile, thanks for pointing out. I have to check this issue a bit then I will be able to let you know.
<br> 
### ans of second para 
I think you didnt get the whole thing properly. There definitely has an index.html file in the owner's site and mine as well, you will find mine in this [link](https://github.com/HabibRh26/HabibRh26.github.io/blob/e45968ea797d055df8a806587150f7fafdfce230/index.html).
<br>
>>`"In your case everything is in one file........"` </br>


I didnt get this line of your's actually, I do not have everything in one file, this repo contains different folders where different things are stored, that's what i understood about this line. 

</br>

>>"Can you tell me how this repo works.........."
" Can you guide me how to use ..........."

**Answer** : <br>
It is not possible to make you understand writing here. But I will try to help you. Initially, I can share some info. I also found it very difficult to understand how any Hugo theme works. But this theme is a bit different than any usual. </br>
**How it works** 
<br> 
this theme has yaml files, html files, js files. You need not to worry about these. Hugo theme actually works like- first when you run the theme by using the yaml files it will generate some html files(in the public folder) through which the website is visible to us. It does not work like the traditional html sites where you run html code directly and it got visiblity. When you run the commands in your local machine it would work if you do everything according to the documentation given in the [main](https://github.com/hossainemruz/toha) repo. What i did was read the docs and ran commands accordingly in the local machine. I found the documentation best than any other online resource on  any topic. So, that is from me initially. So, I think first you need to read and do accordingly. 
<br>
And the problem you are facing after running the main repo theme is because you did not customized anything in the respective folders like you dont have the images in the right location which are in the author's main repo, you didnt also configured the `config.yaml` file correctly I believe. So, correct these issues and I hope it will work for you. Thanks

**Comment by ipankajkumar93** [comment]
`And the problem you are facing after running the main repo theme is because you did not customized anything in the respective folders like you dont have the images in the right location which are in the author's main repo,`

I didn't fork the main repo, I forked this one.
https://github.com/hossainemruz/hossainemruz.github.io

and that's why it is so confusing for me now, coz for Emruz this repository works, without any html files, if I go to  hossainemruz.github.io , I can see his website working just fine and his data is all stored in various yaml files.
He has defined everything in data/sections> and sumhow it works and his repo doesn't even have any html files.

But when I fork his repo all I can see after running my url is default.md file stored in archetype.

It is suc ha nice theme but there is no clear way on how to deploy on Github Pages, your repo has html files which loads on url, Emruz is doing it without any html files.
How ?

**Comment by HabibRh26** [comment]
@ipankajsingh Emruz is not doing it without htmls, there are htmls also, I read the official hugo doc theme how to host any hugo theme on github and successfully hosted it. <br>
### How hugo theme hosting works in GitHub
from the [doc](https://gohugo.io/hosting-and-deployment/hosting-on-github/) it states that you have to create two github repo, the first one would be with a `username.github.io` name and another one which will contain all your website files. First one will work as the live version of your website and any changes in the folder you make would be in the second repo. For example: you will see that Emruz has two repo one is the extension with `github.io` and another named `[toha-example-site](https://github.com/hossainemruz/toha-example-site)`. This also applies to me, I had to create one github.io repo and another repo which I named `[staticsite](https://github.com/HabibRh26/staticSite)` that contains a folder called  `public`. If you click this `public` folder then it will take you to the my `github.io` repo which is responsible for the live verison of my hosted website on github. <br>
**Long Story Short**<br>
If you want to host on GitHub then you have to create a repo in which you will use another repo named `github.io` as a submodule. For better clarification search on youtube, what is submodule in github or seach in a blog forum([dev](https://dev.to/).to) how to host hugo theme on github etc. Hope it is clear now. thanks

**Comment by ipankajkumar93** [comment]
I just looked at what you said, but what you have done is not what Emruz has done.

Your static repo points to your github.io from `public @ e45968e` folder, but if you check Emruz's repo `toha-example-site` there is no public folder pointing to his github.io
He just has a themes folder which points to  `hossainemruz /toha`
Seriously , what is someone like me suppose to do when 2 main contributors have done 2 different things.

and none of Emruz's repository has any html files.
I checked 
1.  `hossainemruz /hossainemruz.github.io` 
2.  `hossainemruz /toha `
3.  `hossainemruz /toha-example-site` 
I am sorry but I just can't see where are html files in these 3 repo ?

His 1st repo is pointing to 2nd repo from themes folder that's it.
and 3rd repo doesn't point to 1st repo at all, infact this one also points to 2nd repo from themes folder.

Where as you are pointing to your github.io from your static repo and Emruz has done nothing like that, I just cloned your github.io repo and I can run it without needing to point to 2nd repo but it's a lot of pain to modify a single index.html which has everything in it.
So what was the point of 2nd repo ?


It's such a great theme but kindly come up with a detailed instructions, otherwise only few people who know about all this can use it but no one else.
I spent 3 days trying everything but nothing happened.

Emruz wrote here
https://toha.netlify.app/posts/deploy-site/github-pages/
that it might take a month or so to write the article but that was 3 months ago.

Disappointed :(

It will be really nice if you could ask him to write an article where we can deploy this website on Github pages with empty sections so that we can modify and do as needful.
I sent him a mail 2 days ago and he told that he'll write an article in 1 day but that also didn't happen.
Or may be I should just leave the hope.
